src/components/Login/Login.js

```diff
@@ -0,0 +1,132 @@
+import React, { useEffect, useState } from 'react';
+import { Box, VStack, FormControl } from 'native-base';
+import { NativeModules } from 'react-native';
+
+import { useToggle } from '../../hooks/useToggle';
+import { Password } from './Password';
+import { Footer } from './Footer';
+import { Username } from './Username';
+import { Loggedin } from './Loggedin';
+import { Header } from './Header';
+
+const { ForgeRockModule } = NativeModules;
+
+// Given a CallbackType, return a Component
+const callbackToComponentMap = {
+  NameCallback: ({ label, setter }) => (
+    <Username setUsername={setter} label={label} key={label} />
+  ),
+  PasswordCallback: ({ label, setter }) => (
+    <Password setPass={setter} label={label} key={label} />
+  ),
+};
+
+function Login({ data, callbacks, navigation }) {
+  const [username, setUsername] = useState('');
```

Comment:
Do we want to leverage a `useReducer` instead for this instead of a `useState` for each and every callback-component pairing? Also, is this where you wanted to pull in a library for assistance?

Reply:
No, no library thoughts here. Could use a useReducer if preferred.